### PR TITLE
Feature/backbutton exit

### PR DESCRIPTION
**Pull request by bmaryam**

### CODE REVIEW

**Comment by claudiiafg** [comment]
I can't test this, so i'll let someone with an android do it

**Comment by bmaryam** [comment]
> Bugs (Might be only on my phone):
> 
> * Opening the app and pressing the back button back right away closes the app without the exit confirmation popup
> * Opening the popovers (point of interest, shuttle bus schedule) and then pressing the back button, displays the exit confirmation instead of closing the popovers
> * Opening the building info popover and then pressing the back button closes the app without displaying the exit confirmation
> * Can't exit the building floor map with the back button anymore (I think it was there before, someone confirm)
> * If we press the back button when the exit confirmation button is in the screen opens a new one.
> * If we open a popover (point of interest or shuttle bus), close it, then click on the building info, the back button does not works anymore.
> 
> Format:
> 
> * Double quotes to single quotes.

On my phone, points 1, 2, 3 and 6 work just fine.
Point 4 is still being looked into
Point 5 is a bug

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,30 +1,69 @@
...
-import { UserServices } from 'src/services/user.services';
-import { TranslationService } from '../services/translation.service';
+import { Platform, AlertController } from "@ionic/angular";
+import { SplashScreen } from "@ionic-native/splash-screen/ngx";
+import { StatusBar } from "@ionic-native/status-bar/ngx";
+import { DataSharingService } from "../services/data-sharing.service";
+import { UserServices } from "src/services/user.services";
+import { TranslationService } from "../services/translation.service";
+import { Router } from "@angular/router";
+import { GoogleMapComponent } from "./components/outdoor/google-map/google-map.component";
```
Comment (alecadub) [inline src/app/app.component.ts]: Nice feature :D 

As angel commented, we are using a single quote for formatting in the entire project :O Check out the wiki for how to change it https://github.com/claudiiafg/MapConU/wiki/Formatting.

I also get the same bug mentionned by angel, but I'm on an emulator

image

The phone is a pixel 2. What is yours @achung98 and what's yours @bmaryam ??

**Comment by bmaryam** [comment]
Enhancements to be done:

- Remove exit popup outdoor when user is looking at outdoor POIs
- Make changes so that the popup doesn't show up more than once when clicking the back button many time

**Pull request merged**